Makefile.am

```diff
@@ -45,6 +45,10 @@ src_libfabric_la_SOURCES = \
 	src/log.c \
 	$(common_srcs)
 
+if MACOS
+src_libfabric_la_SOURCES += src/osx/osd.c
+endif
+
```

Comment:
FWIW, if it makes the Makefile.am a little cleaner, it's ok to add src/osx/osd.c and the relevant OS X headers into this same `if MACOS` conditional.  Automake is smart enough to know that if you add things to a `_SOURCES` macro inside of an AM conditional, that those things must also be added to the tarball.

Also FWIW, `make distcheck` will make a tarball and then expand that tarball and try to build it 2 different ways.  It's a nice sanity check to ensure that you didn't forget to add something to `_SOURCES`, for example.